src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -5820,6 +5820,462 @@ public Derived() : base() { }
+            {
+                public Base()
+                {
+                }
+    
+                public required abstract string Str { get; set; }
+            }
+
+            public class Derived : Base
+            {
+                [SetsRequiredMembers]
+                public Derived(string str) : base()
+                {
+                    this.Str = str;
+                }
+
+                public override required string Str { get; set; }
+            }
+            """;
+
+        var comp = CreateCompilationWithRequiredMembers(code);
+        comp.VerifyDiagnostics();
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/74423")]
+    public void SetsRequiredMembersHonoredForPropertyOverride_06()
+    {
+        var code = """
+            #nullable enable
+            using System.Diagnostics.CodeAnalysis;
+
+            public abstract class Base
+            {
+                public Base()
+                {
+                }
+    
+                public required abstract string Str { get; set; }
+            }
+
+            public class Derived : Base
+            {
+                public Derived() : base()
+                {
+                }
+
+                public override required string Str { get; set; }
+            }
+
+            public class DerivedDerived : Derived
+            {
+                [SetsRequiredMembers]
+                public DerivedDerived(string str) : base()
+                {
+                }
+
+                public override required string Str { get; set; }
+            }
+            """;
+
+        var comp = CreateCompilationWithRequiredMembers(code);
+        comp.VerifyDiagnostics(
+            // (25,12): warning CS8618: Non-nullable property 'Str' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable.
+            //     public DerivedDerived(string str) : base()
+            Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "DerivedDerived").WithArguments("property", "Str").WithLocation(25, 12),
+            // (25,12): warning CS8618: Non-nullable property 'Str' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable.
+            //     public DerivedDerived(string str) : base()
+            Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "DerivedDerived").WithArguments("property", "Str").WithLocation(25, 12)
+        );
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/74423")]
+    public void SetsRequiredMembersHonoredForPropertyOverride_07()
+    {
+        var code = """
+            #nullable enable
+            using System.Diagnostics.CodeAnalysis;
+
+            public abstract class Base
+            {
+                public Base()
+                {
+                }
+    
+                public required abstract string Str { get; set; }
```

Comment:
Maybe we should also test some similar cases to this where required is missing on the base (which appears to be permitted)? The results might further validate the need for the logic added in this PR.

Reply:
> which appears to be permitted

It is not permitted.

Reply:
I didn't see any error in the following scenario: [SharpLab](https://sharplab.io/#v2:EYLgtghglgdgPgAQEwEYCwAoBBmABM3AYVwG9NcL88A3KAJwBcBXCAG3xQAZcAFOgewAOpXAHMApgwDcuAM6SZAX0zKMmHPiS4AIuLpRq4gCa4QRUuUoa64gI5N6x3P0N19R8R258hIidLkFXFVFIA==).

```cs
using System;
public class C {
    public virtual string Prop { get; set; }
}

public class Derived : C {
    public required override string Prop { get; set; }
}
```

Was there something I misunderstood here?

Reply:
Apparently I don't remember my own feature...